lib/spack/spack/installer.py

```diff
@@ -890,28 +899,21 @@ def traverse_dependencies(self, spec=None, visited=None) -> Iterator["spack.spec
 class BuildTask:
     """Class for representing the build task for a package."""
 
-    def __init__(
-        self,
-        pkg: "spack.package_base.PackageBase",
-        request: Optional[BuildRequest],
-        compiler: bool,
-        start: float,
-        attempts: int,
-        status: str,
-        installed: Set[str],
-    ):
+    def __init__(self, pkg: "spack.package_base.PackageBase", **kwargs):
```

Comment:
Let's not use `**kwargs`, where passing the wrong argument is not a `TypeError`. 

What I meant in the other PR was [keyword only arguments](https://peps.python.org/pep-3102/):
```suggestion
    def __init__(
        self, 
        pkg: "spack.package_base.PackageBase", 
        *,
        request: Optional[BuildRequest],
        compiler: bool,
        start: float,
        attempts: int,
        status: str,
        installed: Set[str],
    ):
```

Reply:
Ah.  Thanks for the clarification.

Reply:
BTW `request` is not optional so I cleaned that up as well.